### ISSUE TITLE
Code not working

### ISSUE THREAD

**Issue by mtliou** [kyegomez/tree-of-thoughts#93]
This code can't solve simple 24 games like: "4 4 6 8" Is the code working for anyone else?

**Comment by kyegomez** [comment]
@mtliou hey try again please, I just shipped a new update

**Comment by moht355** [comment]
Yeah idk what you're doing but it's not working... I've spent so much time on this but I'm now seeing all the negative comments regarding the code. Has it ever worked for anyone?

"2024-09-07T16:59:05.621977-0400 Finished DFS for state: 

Your task: is to use 4 numbers and basic arithmetic operations (+-*/) to obtain 24 in 1 equation, return only the math


{
    "final_thoughts": [
        {
            "thought": "One common set of numbers used in these types of problems is 8, 3, 3, and 1. Let's see if we can achieve 24 using these numbers: \\(8 \\\\times (3 - (3 - 1)) = 8 \\\\times 1 = 8\\), which doesn't work. Let's try another approach.\n\nAnother set is 8, 3, 2, and 1: \\(8 \\\\times (3 - (2 - 1)) = 8 \\\\times 2 = 16\\), which also doesn't work. However, using 8, 3, 2, and 1 but in a different operation: \\((8 \\\\div (3 - 1)) \\\\times 3 = (8 \\\\div 2) \\\\times 3 = 4 \\\\times 3 = 12\\), which still doesn't work. I need to re-evaluate my approach to achieve exactly 24.\\n\nLet's try another set: 4, 4, 4, 1: \\((4 + 4) \\\\times (4 - 1) = 8 \\\\times 3 = 24\\). This seems to work.",
            "evaluation": 0.9
        },
        {
            "thought": "One possible solution using the numbers 8, 3, 3, and 1 is: \\( (8 - 3) \\times (3 + 1) = 24 \\).",
            "evaluation": 0.9
        },
        {
            "thought": "To solve the problem of obtaining 24 using four numbers and basic arithmetic operations, I need to first identify the four numbers. Since the numbers are not provided, I will assume a common set of numbers often used in such puzzles, such as 1, 3, 4, and 6. The goal is to combine these numbers using addition, subtraction, multiplication, and division to reach exactly 24. A common solution for this set is: (6 / (1 - (3 / 4))) = 24. This equation uses division and subtraction to manipulate the numbers to achieve the target number.",
            "evaluation": 0.9
        },
        {
            "thought": "One possible equation using the numbers 8, 3, 3, and 1 to obtain 24 is: (8 / (3 - (1 / 3))) = 24. This involves dividing 8 by the result of the subtraction of 1/3 from 3, which simplifies to 24.",
            "evaluation": 1.0
        }
    ],
    "pruned_branches": [],
    "highest_rated_thought": {
        "thought": "One possible equation using the numbers 8, 3, 3, and 1 to obtain 24 is: (8 / (3 - (1 / 3))) = 24. This involves dividing 8 by the result of the subtraction of 1/3 from 3, which simplifies to 24.",
        "evaluation": 1.0
    }
}"

**Comment by kyegomez** [comment]
> Yeah idk what you're doing but it's not working... I've spent so much time on this but I'm now seeing all the negative comments regarding the code. Has it ever worked for anyone?
> 
> 
> 
> "2024-09-07T16:59:05.621977-0400 Finished DFS for state: 
> 
> 
> 
> Your task: is to use 4 numbers and basic arithmetic operations (+-*/) to obtain 24 in 1 equation, return only the math
> 
> 
> 
> 
> 
> {
> 
>     "final_thoughts": [
> 
>         {
> 
>             "thought": "One common set of numbers used in these types of problems is 8, 3, 3, and 1. Let's see if we can achieve 24 using these numbers: \\(8 \\\\times (3 - (3 - 1)) = 8 \\\\times 1 = 8\\), which doesn't work. Let's try another approach.\n\nAnother set is 8, 3, 2, and 1: \\(8 \\\\times (3 - (2 - 1)) = 8 \\\\times 2 = 16\\), which also doesn't work. However, using 8, 3, 2, and 1 but in a different operation: \\((8 \\\\div (3 - 1)) \\\\times 3 = (8 \\\\div 2) \\\\times 3 = 4 \\\\times 3 = 12\\), which still doesn't work. I need to re-evaluate my approach to achieve exactly 24.\\n\nLet's try another set: 4, 4, 4, 1: \\((4 + 4) \\\\times (4 - 1) = 8 \\\\times 3 = 24\\). This seems to work.",
> 
>             "evaluation": 0.9
> 
>         },
> 
>         {
> 
>             "thought": "One possible solution using the numbers 8, 3, 3, and 1 is: \\( (8 - 3) \\times (3 + 1) = 24 \\).",
> 
>             "evaluation": 0.9
> 
>         },
> 
>         {
> 
>             "thought": "To solve the problem of obtaining 24 using four numbers and basic arithmetic operations, I need to first identify the four numbers. Since the numbers are not provided, I will assume a common set of numbers often used in such puzzles, such as 1, 3, 4, and 6. The goal is to combine these numbers using addition, subtraction, multiplication, and division to reach exactly 24. A common solution for this set is: (6 / (1 - (3 / 4))) = 24. This equation uses division and subtraction to manipulate the numbers to achieve the target number.",
> 
>             "evaluation": 0.9
> 
>         },
> 
>         {
> 
>             "thought": "One possible equation using the numbers 8, 3, 3, and 1 to obtain 24 is: (8 / (3 - (1 / 3))) = 24. This involves dividing 8 by the result of the subtraction of 1/3 from 3, which simplifies to 24.",
> 
>             "evaluation": 1.0
> 
>         }
> 
>     ],
> 
>     "pruned_branches": [],
> 
>     "highest_rated_thought": {
> 
>         "thought": "One possible equation using the numbers 8, 3, 3, and 1 to obtain 24 is: (8 / (3 - (1 / 3))) = 24. This involves dividing 8 by the result of the subtraction of 1/3 from 3, which simplifies to 24.",
> 
>         "evaluation": 1.0
> 
>     }
> 
> }"

What's not working? Can you be more specific? It seems to be working just fine???
